File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/core/RowCodecBasic.kt
package dev.typr.foundations.docs.core

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*
import java.time.Instant

@Suppress("unused")
class RowCodecBasic {
    //start
    data class Person(val id: Int, val name: String, val createdAt: Instant)

    val personCodec: RowCodec<Person> =
        RowCodec.builder<Person>()
            .field(PgTypes.int4, Person::id)
            .field(PgTypes.text, Person::name)
            .field(PgTypes.timestamptz, Person::createdAt)
            .build(::Person)
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/RowCodecBasic.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*


import java.time.Instant

@SuppressWarnings(Array("unused"))
object RowCodecBasic:
  //start
  case class Person(id: Int, name: String, createdAt: Instant)

  val personCodec: RowCodec[Person] = RowCodec.builder[Person]()
    .field(PgTypes.int4)(_.id)
    .field(PgTypes.text)(_.name)
    .field(PgTypes.timestamptz)(_.createdAt)
    .build(Person.apply)
  //stop
